src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -845,8 +867,6 @@ public void Close()
 
 			if (settings.IsQueryCacheEnabled)
 			{
-				queryCache.Destroy();
-
 				foreach (var cache in queryCaches.Values)
```

Comment:
No need anymore to explicitly destroy `queryCache`, it is now into `queryCaches`.

(Comment re-added, it has been lost in last rebase.)